### ISSUE TITLE
since introducing training multiple subjects, this repo and shiv repo gets bad results

### ISSUE THREAD

**Issue by 1blackbar** [TheLastBen/fast-stable-diffusion#231]
Me and other people noticed that new code introduced in last 2 days now makes the results worse than it was before , something happened maybe in the main diffusers branch, its hard to tell what but the difference is noticeable, subjects likeness is all over the place, fatter faces and not really that close, its pretty weird

**Comment by x02Sylvie** [comment]
Maybe it's related to change with learning rate
        " -   --learning_rate=2e-6 \\\n",
        "+    --learning_rate=1e-6 \\\n",

**Comment by TheLastBen** [comment]
the text encoder training doesn't make thing worse, it make things more efficient, you don't need 4000 steps anymore, you only need 1600 steps, if you go over it you will overtrain it

I trained it on 2 subjects : 

![197557015-e5404f61-f4e9-403d-91d3-8e1dfd4c1f34](https://user-images.githubusercontent.com/110583491/197674279-056e898c-f2cb-4316-8601-9eed46fe9338.png)
![download (3)](https://user-images.githubusercontent.com/110583491/197674293-6722d979-3739-4f59-85e6-6a6bfb5748bc.png)
![download (4)](https://user-images.githubusercontent.com/110583491/197674307-1fc93c65-2984-4e67-b20d-273cd7c1598e.png)
![download (6)](https://user-images.githubusercontent.com/110583491/197674319-38828da9-12a8-4d9e-83d9-8474b4bd0ed4.png)
![download (7)](https://user-images.githubusercontent.com/110583491/197674330-cbc28f62-e4cb-4079-b20b-5f691b34b318.png)

**Comment by TheLastBen** [comment]
New better, faster method incoming, it will be ready in a few hour

**Comment by 1blackbar** [comment]
well, i keep track of overtraining on shiv repo easily by looking at the 4 samples after model gets saved each 400 steps until 3200 and it takes like 10 seconds to see if the likeness and stylisation are ready, on this repo to do that and run the webui to test models is taking a lot of time.
I noticed on his repo with 20 imgs and 2400 steps i get amost overfitting but still very stylisable , so thats the sweet spot for me atm with his rate, but with this repo i used 1700 and it did not work with all subjects so im still looking for best settings.
Especialy now that its 1e-6 and not 2e-6.

**Comment by TheLastBen** [comment]
use the new method

**Comment by 1blackbar** [comment]
I just did 1500 and its not enough for 2 people , maybe i shoud double the steps so if i used 1500 i should go for 3000 ? but the results are not really good, deformed faces.
Defaults definitely dont work for me, why its suddenly just 600 steps ? This is just not enough

**Comment by TheLastBen** [comment]
I'm saying it's good because I tried it 
the first two are the default ones and the last two are trained with 600 steps 

![or 600s](https://user-images.githubusercontent.com/110583491/197886700-d9837f3e-6360-4d44-bde4-a21d31a60fbb.png)
![600s orem](https://user-images.githubusercontent.com/110583491/197886795-cfaf392d-d286-4d1c-a790-b3c06f3db8e5.png)

![600s](https://user-images.githubusercontent.com/110583491/197886658-f2b2affc-7814-4ffb-b58b-ef8d0f1a7b0d.png)

![600s d](https://user-images.githubusercontent.com/110583491/197886625-86f4096e-515f-4a42-85f2-c044d3c599ed.png)

**Comment by 1blackbar** [comment]
These look very good, 30 imgs each and 600 steps ? 
Theres little room for error if i dont change defauts 1e-6 , my files are named person_man_krb(number).jpg

**Comment by TheLastBen** [comment]
"person_man_krb(number).jpg"
that's the problem, with that name, you're destroying the class person and the class man. you must use only the instance name in the files krb(1).jpg .... krb(2).jpg .... etc

**Comment by TheLastBen** [comment]
the whole method is based on the images filenames which should be only one random unknown keyword

**Comment by toyxyz** [comment]
I trained 3 subjects at x1500 steps for each subject.

![image](https://user-images.githubusercontent.com/8006000/198540142-c3bdedcd-b0b8-4906-a7df-d696331552e7.png)
